server2/routes/confirmationRouter.js

```diff
@@ -0,0 +1,51 @@
+var express = require("express");
+var router = express.Router();
+var User = require("../models/user");
+var EmailToken = require("../models/emailToken");
+
+router
+    .route("/:token")
+    .get((req, res, next) => {
+        try {
+            const getToken = () => {
+                return new Promise((resolve, reject) => {
+                    resolve(EmailToken.findOne());
+                })
+            };
+
+            const verify = async () => {
+                const tokenObj = await getToken();
+                const token = tokenObj.token;
+
+                const jsonReceived = decodeURIComponent(req.params.token);
+                const unpacked = JSON.parse(jsonReceived);
+
+                const recToken = unpacked.token;
+
+                if (recToken == token) {        
+                    console.log("Token confirmed.");
+                    User.create(unpacked.user);
+                    
+                    // Confirm successful page
+                    res.send("Your Email is confirmed!");
+                } else {
+                    console.error("Token does not match!");
+                    // Confirm failed page
+                    res.send("Confirmation failed!\n" +
+                        "Some unexpected error has occured:\n\t"
+                        + err);
+                }
+                EmailToken.destroy(
```

Comment:
why token mismatch needs to destroy the token? Considering this implementation, if i am an attacker, i can always flush this api (e.g with 10 - 100 qps, which is very easy to manipulate in reality) by inputting an empty string in the request token field. As such, your implementation will always try to clear the token from the db and hence no user can successfully register even they receive a valid link and none of them know why it fails.